enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,590 @@
+    // defaultConfig.
+    Config ConfigReferent `json:"config"`
+
+    // lastObservedGeneration is the observed generation of the add-on configuration.
+    LastObservedGeneration int64 `json:"lastObservedGeneration"`
+}
+```
+
+As a specific case, we want to introduce a `AddOnDeployementConfig` API in cluster scope to configure the add-on
+deployment.
+
+The add-on developers can choose to use this API to configure their add-on deployment or use their own API for
+configuration.
+
+```golang
+// AddOnDeployementConfig represents a deployment configuration for an add-on.
+// AddOnDeployementConfig is a cluster-scoped resource.
+type AddOnDeployementConfig struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // spec represents a desired configuration for an add-on.
+    // +required
+    Spec AddOnDeployementConfigSpec `json:"spec"`
+}
+
+type AddOnDeployementConfigSpec struct {
+    // CustomizedVariables is a list of name-value variables for the current add-on deployment.
+    // The add-on implementation can use these variables to render the add-on deployment.
+    // The default is an empty list.
+    // +optional
+    CustomizedVariables []CustomizedVariable `json:"customizedVariable,omitempty"`
+
+    // NodePlacement enables explicit control over the scheduling of the add-on.
+    // +optional
+    NodePlacement NodePlacement `json:"nodePlacement,omitempty"`
+}
+
+// CustomizedVariable represents a customized variable for add-on deployment.
+type CustomizedVariable struct {
+    // Name of this variable. Must be a C_IDENTIFIER.
+    Name string `json:"name"`
+
+    // Value of this variable.
+    Value string `json:"value"`
+}
+
+// NodePlacement describes node scheduling configuration for the pods.
+type NodePlacement struct {
+    // NodeSelector defines which Nodes the Pods are scheduled on. The default is an empty list.
+    // +optional
+    NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+    // Tolerations is attached by pods to tolerate any taint that matches
+    // the triple <key,value,effect> using the matching operator <operator>.
+    // The default is an empty list.
+    // +optional
+    Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+}
+```
+
+Examples:
+
+1. Using `AddOnDeployementConfig` to configure add-on
+
+    User have a default add-on deployment configuration on the hub
+
+    ```yaml
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: AddOnDeployementConfig
+    metadata:
+      name: addon-default-placement
+    spec:
+      nodeSelector:
+        "kubernetes.io/arch": "amd64"
+      tolerations:
+      - key: "node.kubernetes.io/unreachable",
+        operator: "Exists",
+        effect: "NoSchedule"
+    ```
+
+    User set up the configuration referance in the `ClusterManagementAddOn` to share this configuration to their add-ons
```

Comment:
How does the user know that the addon supports this configuration type?

Reply:
good catch, I think the ClusterManagementAddOn should be applied along with addon hub controller deployed, so users should not need to configure this, I will make this clear